core/store/migrate/migrate.go

```diff
@@ -99,12 +101,15 @@ func ensureMigrated(ctx context.Context, db *sql.DB) error {
 	})
 }
 
-func Migrate(ctx context.Context, db *sql.DB) error {
+func Migrate(ctx context.Context, db *sql.DB, healthReportPort uint16, lggr logger.Logger) error {
 	if err := ensureMigrated(ctx, db); err != nil {
 		return err
 	}
 	// WithAllowMissing is necessary when upgrading from 0.10.14 since it
 	// includes out-of-order migrations
+	ibhr := services.NewInBackupHealthReport(healthReportPort, lggr)
+	ibhr.Start()
+	defer ibhr.Stop()
```

Comment:
Actually, maybe skip several callers. I think it is important to include as much of the startup time as possible, plus this will avoid the leaking in to the administrative commands that don't need to be reporting health in the first place, like `ResetDatabase`.